apps/settings/js/modules/addon_manager.js

```diff
@@ -205,71 +205,114 @@ define(function(require) {
-          // XXX: Will themes have a role, and should we ignore them too?
-          if (manifest.role === 'addon') {
-            return;
-          }
+      // Ignore apps that are themselves add-ons
+      // XXX: Will themes have a role, and should we ignore them too?
+      if (manifest.role === 'addon') {
+        return Promise.resolve();
+      }
 
-          // If the addon doesn't have high enough privileges to affect this app
-          // then we can just return now.
-          if (!this._privilegeCheck(addon, app)) {
-            return;
-          }
+      // If the addon doesn't have high enough privileges to affect this app
+      // then we can just return now.
+      if (!this._privilegeCheck(addon, app)) {
+        return Promise.resolve();
+      }
 
-          // Get the URL of the app origin plus its launch path to
-          // compare against each of the filters.
-          // XXX: Note that we and do not check the paths used by
-          // activity handlers or any other paths in the manifest.
-          var launchPath = manifest.launch_path || '';
-          var launchURL = new URL(launchPath, app.origin).href;
-          var appliedCustomizations = [];
-
-          // Now loop through the filters to see what customizations are
-          // applied to this app
-          for(var i = 0; i < filters.length; i++) {
-            var filter = filters[i];
-            if (filter.test(launchURL)) {
-              appliedCustomizations.push(customizations[i]);
-              break;
-            }
-          }
+      // Get the URL of the app origin plus its launch path to
+      // compare against each of the filters.
+      // XXX: Note that we and do not check the paths used by
+      // activity handlers or any other paths in the manifest.
+      var launchPath = manifest.launch_path || '';
+      var launchURL = new URL(launchPath, app.origin).href;
+
+      // For each customization, compile the filter string into a regexp
+      var filters = customizations.map(customization =>
+        new RegExp(customization.filter));
+
+      var appliedCustomizations = [];
+
+      // Now loop through the filters to see what customizations are
+      // applied to this app
+      for(var i = 0; i < filters.length; i++) {
+        var filter = filters[i];
+        if (filter.test(launchURL)) {
+          appliedCustomizations.push(customizations[i]);
+          break;
+        }
+      }
 
-          // If any customizations were applied to this app, then add it
-          // to the list of customizedApps
-          if (appliedCustomizations.length > 0) {
-            customizedApps.push({
-              app: app,
-              customizations: appliedCustomizations
-            });
-          }
+      // If any customizations were applied to this app, return the list of
+      // appliedCustomizations along with the app
+      if (appliedCustomizations.length > 0) {
+        return Promise.resolve({
```

Comment:
I think we can resolve with only `applideCustomizations` as `app` and `addon` are passed as parameters.